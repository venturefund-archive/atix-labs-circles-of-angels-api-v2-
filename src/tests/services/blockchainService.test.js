const axios = require('axios');
const blockchainService = require('../../rest/services/blockchainService');
const { txTypes } = require('../../rest/util/constants');
const { API_RESPONSE } = require('../externalApiResponse.mock');

jest.mock('../../rest/services/tokenService', () => ({
  getTokenBySymbol: jest.fn(() =>
    Promise.resolve({
      id: 1,
      name: 'ETC',
      symbol: 'ETC',
      decimals: 18
    })
  )
}));

jest.mock('axios');

describe('Testing blockchainService', () => {
  const ADDRESS = '0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2';
  axios.get.mockResolvedValue(API_RESPONSE);
  describe('Testing getTransactions', () => {
    it('Should return all sent transactions', async () => {
      const response = await blockchainService.getTransactions({
        currency: 'ETC',
        address: ADDRESS,
        type: txTypes.SENT
      });
      expect(response.transactions).toMatchSnapshot();
      expect(
        response.transactions.every(transaction => transaction.value !== '0')
      ).toBeTruthy();
      expect(
        response.transactions.every(transaction => transaction.from === ADDRESS)
      ).toBeTruthy();
    });

    it('Should return all received transactions', async () => {
      const response = await blockchainService.getTransactions({
        currency: 'ETC',
        address: ADDRESS,
        type: txTypes.RECEIVED
      });
      expect(response.transactions).toMatchSnapshot();
      expect(
        response.transactions.every(transaction => transaction.value !== '0')
      ).toBeTruthy();
      expect(
        response.transactions.every(transaction => transaction.to === ADDRESS)
      ).toBeTruthy();
    });
  });
});
