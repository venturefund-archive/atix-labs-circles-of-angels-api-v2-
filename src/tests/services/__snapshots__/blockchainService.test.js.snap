// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing blockchainService Testing getTransactions Should return all received transactions 1`] = `
Array [
  Object {
    "from": "0xcfc0f98f30742b6d880f90155d4ebb885e55ab33",
    "timestamp": "Nov-08-2022 05:22:59 PM+UTC",
    "to": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "tokenSymbol": "ETC",
    "txHash": "0x21c707b07f370bb51243d648bb91cafd17cdfed99e434f2fc47855d6b01f4846",
    "value": 0.0289815,
  },
  Object {
    "from": "0xd108fd0e8c8e71552a167e7a44ff1d345d233ba6",
    "timestamp": "Nov-03-2022 04:24:23 AM+UTC",
    "to": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "tokenSymbol": "ETC",
    "txHash": "0x4de4d8805d120430ec7fcea5b1ffc8a6235da8dfa2ff20753ffc3bb86c667a8b",
    "value": 0.6111,
  },
  Object {
    "from": "0xaf3de19c493cfe41f4995c3ab6aece11337e007f",
    "timestamp": "Oct-31-2022 10:53:59 AM+UTC",
    "to": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "tokenSymbol": "ETC",
    "txHash": "0x3487488ea9885d29db5c93384542a02b32fa5ca45082c0994bc13eae540da364",
    "value": 0.0323900902192067,
  },
  Object {
    "from": "0xaf3de19c493cfe41f4995c3ab6aece11337e007f",
    "timestamp": "Oct-31-2022 10:47:59 AM+UTC",
    "to": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "tokenSymbol": "ETC",
    "txHash": "0x231c9f97110097d478713b458fe398678caf746396caaea61650fb640e15f3ef",
    "value": 0.02778,
  },
  Object {
    "from": "0xaf3de19c493cfe41f4995c3ab6aece11337e007f",
    "timestamp": "Oct-31-2022 03:48:47 AM+UTC",
    "to": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "tokenSymbol": "ETC",
    "txHash": "0xd52fb5efa80bb942aa712d46744ff2c7b91c6b3098fec0b49a859c0a79d85ad7",
    "value": 0.03727,
  },
  Object {
    "from": "0xaf3de19c493cfe41f4995c3ab6aece11337e007f",
    "timestamp": "Sep-09-2022 07:31:23 PM+UTC",
    "to": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "tokenSymbol": "ETC",
    "txHash": "0xdc97994894ff50362f9e945d67c1474d67405df44e9cbca0f06d04faa6e4f59d",
    "value": 0.00467,
  },
  Object {
    "from": "0x3cd751e6b0078be393132286c442345e5dc49699",
    "timestamp": "Jul-28-2022 08:32:52 PM+UTC",
    "to": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "tokenSymbol": "ETC",
    "txHash": "0xff094645c94b2479a5a164f0cd8f9d705862374ec72340136b67c6dcdb5c57bb",
    "value": 0.01421629,
  },
]
`;

exports[`Testing blockchainService Testing getTransactions Should return all sent transactions 1`] = `
Array [
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Nov-22-2022 04:00:59 PM+UTC",
    "to": "0xa36799739db0db84b500afdd1695f6a670c244f0",
    "tokenSymbol": "ETC",
    "txHash": "0xf9216a75da3c8ecd4857020c4adcea405269850614b5e68c01f798fec76f3b2e",
    "value": 0.01,
  },
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Nov-20-2022 03:32:23 PM+UTC",
    "to": "0xebd26f192a46f1e7b3e2b208922a2f564c1b874e",
    "tokenSymbol": "ETC",
    "txHash": "0x251fe32d87fe995c3af1f15a18ab5694eaf21146dd0fac75f4e3f5eecd34f72b",
    "value": 0.05,
  },
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Nov-18-2022 09:17:35 PM+UTC",
    "to": "0xaf3de19c493cfe41f4995c3ab6aece11337e007f",
    "tokenSymbol": "ETC",
    "txHash": "0xb545a7e786b09e4e694ab949cada88974a4eb23e76b02930fbbdd4a4007ceaf8",
    "value": 0.00165,
  },
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Nov-18-2022 09:15:35 PM+UTC",
    "to": "0xaf3de19c493cfe41f4995c3ab6aece11337e007f",
    "tokenSymbol": "ETC",
    "txHash": "0x2a50df6534452b3c35e089e809d179002493e447c162513d2af458d818f148ca",
    "value": 0.00165,
  },
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Nov-18-2022 07:41:23 PM+UTC",
    "to": "0xaf3de19c493cfe41f4995c3ab6aece11337e007f",
    "tokenSymbol": "ETC",
    "txHash": "0x0bff4f8125be8d763d30f4eccf7a464c57512c6cc6128593003c1a1c56dcb363",
    "value": 0.0083,
  },
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Nov-10-2022 06:32:23 AM+UTC",
    "to": "0x68b3465833fb72a70ecdf485e0e4c7bd8665fc45",
    "tokenSymbol": "ETC",
    "txHash": "0x3a2c5ab0fbb56e574bf56f687b178a1b7fba2846dcf545ee09ba6e56c46dd007",
    "value": 0.058813809537081194,
  },
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Nov-10-2022 03:48:59 AM+UTC",
    "to": "0x68b3465833fb72a70ecdf485e0e4c7bd8665fc45",
    "tokenSymbol": "ETC",
    "txHash": "0x79967ad98784696bc9c711222a6edffb71078672d45f76e18f45539503d0ee80",
    "value": 0.016537203834757774,
  },
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Nov-08-2022 05:10:11 PM+UTC",
    "to": "0x68b3465833fb72a70ecdf485e0e4c7bd8665fc45",
    "tokenSymbol": "ETC",
    "txHash": "0xb6d94ea39ddc66351d122bf8018e56c352eeecbbb0b9fd3111c06a755f611166",
    "value": 0.013335187379124255,
  },
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Nov-07-2022 09:55:35 PM+UTC",
    "to": "0x68b3465833fb72a70ecdf485e0e4c7bd8665fc45",
    "tokenSymbol": "ETC",
    "txHash": "0xa395f5d096a978cc1fdf7787a66ea73e3d09ecb76082e3a6a18f954d98db3bc2",
    "value": 0.01095109139850324,
  },
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Nov-04-2022 03:47:47 PM+UTC",
    "to": "0xaf3de19c493cfe41f4995c3ab6aece11337e007f",
    "tokenSymbol": "ETC",
    "txHash": "0xf4efa8afb2fa8e6cd5dd5391bcde81741f8b56ea307d9fd1d6d2251bcb9972f2",
    "value": 0.00122,
  },
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Nov-04-2022 03:42:47 PM+UTC",
    "to": "0xaf3de19c493cfe41f4995c3ab6aece11337e007f",
    "tokenSymbol": "ETC",
    "txHash": "0xbd8ac0cd4348a43abdccb313c65c167891d6a43370ec4d94fdd3dff6f64a447a",
    "value": 0.00488,
  },
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Nov-04-2022 03:35:47 PM+UTC",
    "to": "0xaf3de19c493cfe41f4995c3ab6aece11337e007f",
    "tokenSymbol": "ETC",
    "txHash": "0xb16c0c984106c14309659de8e96339ef98ec1d6a2819bc0a92f351c07cb5dffe",
    "value": 0.00364,
  },
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Nov-03-2022 04:29:23 AM+UTC",
    "to": "0x68b3465833fb72a70ecdf485e0e4c7bd8665fc45",
    "tokenSymbol": "ETC",
    "txHash": "0x035a282f61060ec9a50a6c5279e2457b52be6a17054f225007cd1cd41585b48f",
    "value": 0.6005360547464469,
  },
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Oct-31-2022 10:51:35 AM+UTC",
    "to": "0xaf3de19c493cfe41f4995c3ab6aece11337e007f",
    "tokenSymbol": "ETC",
    "txHash": "0xca9cf159ce55c3b22b649402b3a17aa44aa39f04a382a5cd3f9abb52603fb843",
    "value": 0.03306,
  },
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Oct-31-2022 04:21:35 AM+UTC",
    "to": "0xaf3de19c493cfe41f4995c3ab6aece11337e007f",
    "tokenSymbol": "ETC",
    "txHash": "0x537a402510ab70c2fc120502e30d5b2a66940fa872de48bf111baa3e8551e0d3",
    "value": 0.03153,
  },
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Sep-06-2022 09:12:12 PM+UTC",
    "to": "0xaf3de19c493cfe41f4995c3ab6aece11337e007f",
    "tokenSymbol": "ETC",
    "txHash": "0x80cbf025e2eb9d4262a3854eea6ad8b8675810130d12f237a7bc717c27e013ed",
    "value": 0.066344422329637,
  },
  Object {
    "from": "0x166c8dbcd7447c1fcd265130d3d278d47a3bc7b2",
    "timestamp": "Jul-29-2022 08:26:21 PM+UTC",
    "to": "0xaf3de19c493cfe41f4995c3ab6aece11337e007f",
    "tokenSymbol": "ETC",
    "txHash": "0xd98304a8ca9f21d81739d4093a1bfc95cc9548a54ef4f0e32de4760f221ba428",
    "value": 0.010441652913656231,
  },
]
`;
